libs/ui-lib/lib/ocm/components/clusterConfiguration/ExternalPlatformDropdown.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+import { Dropdown, DropdownItem, DropdownToggle, FormGroup } from '@patternfly/react-core';
+import { CaretDownIcon } from '@patternfly/react-icons';
+import { useField } from 'formik';
+import {
+  NUTANIX_CONFIG_LINK,
+  PlatformType,
+  VSPHERE_CONFIG_LINK,
+  getFieldId,
+} from '../../../common';
+
+const INPUT_NAME = 'platform';
+const fieldId = getFieldId(INPUT_NAME, 'input');
+export type ExternalPlatformType = Extract<PlatformType, 'none' | 'nutanix' | 'oci' | 'vsphere'>;
+
+type ExternalPlatformDropdownProps = {
+  showOciOption: boolean;
+  selectedPlatform: ExternalPlatformType;
+  disabledOciTooltipContent: React.ReactNode;
+  isOciDisabled: boolean;
+  onChange: (selectedPlatform: ExternalPlatformType) => void;
+  dropdownIsDisabled: boolean;
+};
+
+export type ExternalPlatformInfo = {
+  label: string;
+  href: string;
+  tooltip: string;
+};
+
+export const externalPlatformTypes: Record<ExternalPlatformType, ExternalPlatformInfo> = {
+  none: {
+    label: 'None',
+    href: '',
+    tooltip: '',
+  },
+  nutanix: {
+    label: 'Nutanix',
+    href: NUTANIX_CONFIG_LINK,
+    tooltip: '',
+  },
+  oci: {
+    label: 'Oracle  (Requires a custom manifest)',
+    href: '',
+    tooltip:
+      "To integrate with an external partner (for example, Oracle Cloud), you'll need to provide a custom manifest.",
+  },
+  vsphere: {
+    label: 'vSphere',
+    href: VSPHERE_CONFIG_LINK,
+    tooltip: '',
+  },
+};
+
+export const ExternalPlatformDropdown = ({
+  showOciOption,
+  selectedPlatform,
+  disabledOciTooltipContent,
+  isOciDisabled,
+  onChange,
+  dropdownIsDisabled,
+}: ExternalPlatformDropdownProps) => {
+  const [field, { value }, { setValue }] = useField<string>(INPUT_NAME);
+  const [currentPlatform, setCurrentPlatform] = React.useState(
+    externalPlatformTypes[selectedPlatform].label,
+  );
+  const [isOpen, setOpen] = React.useState(false);
```

Comment:
When opening the dropdown, a large console error appears.

Reply:
Yes, I see. It's because we add the "Learn more" link in the dropdown items. @nirfarkas can we do in another way? Maybe putting the link below the dropdown?

Reply:
What's the error?

Reply:
The problem is that patternfly translates dropdown items like elements. And we try to put it inside another and this is an error.

The error:
"react-dom.development.js:67 Warning: validateDOMNesting(...): cannot appear as a descendant of ."

Reply:
Looks like the `DropdownItem` can take a property `component` which defaults to `a`. Maybe you can use an span instead and keep the link for the tooltip?

Alternatively, maybe you can change the `a` to be a Button of type link? Which is probably preferable as we should use more Patterfly components and less ad-hoc code.